.github/workflows/test.yaml

```diff
@@ -0,0 +1,32 @@
+name: Test
+
+on:
+  push:
+    branches:
+      - '*'
+      - '!gh-pages'
+  pull_request:
+    branches:
+      - '*'
+      - '!gh-pages'
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.13.3
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Get dependencies
+        run: make vendor
+      - name: Test
+        run: |
+          make generate
+          go mod tidy
+          (set +e; git status --porcelain | grep '.*'; test $? -eq 1)
+          make lint
+          make test
```

Comment:
We should run in this order

- `make lint`
- `make test`
- `make generate; go mod tidy; (grep ...)`

And failing one should not prevent following commands from running. However, any failure should fail the whole job.